### PR TITLE
Refactor partitioner to use fewer dicts and better names

### PR DESCRIPTION
**Pull request by inducer**
This also represents the partial order among parts in the partition structure.

### CODE REVIEW

**Comment by inducer** [comment]
These changes were bitten out of #148 and aremeant to be squashed.

cc @matthiasdiener

File: pytato/visualization.py [hunk 1]
```diff
@@ -257,22 +258,22 @@ def get_dot_graph(result: Union[Array, DictOfNamedArrays]) -> str:
...
 
-def get_dot_graph_from_partitions(parts: GraphPartitions) -> str:
+def get_dot_graph_from_partition(partition: GraphPartition) -> str:
     r"""Return a string in the `dot <https://graphviz.org>`_ language depicting the
     graph of the computation of *result*.
 
     :arg result: Outputs of the computation (cf.
         :func:`pytato.generate_loopy`).
     """
     # Maps each partition to a dict of its arrays with the node info
```
Comment (matthiasdiener) [inline pytato/visualization.py:268]:
```suggestion
    r"""Return a string in the `dot <https://graphviz.org>`_ language depicting the
    graph of the partitioned computation of *partition*.

    :arg partition: Partitioned computation (cf.
        :func:`pytato.find_partition`).
    """
    # Maps each partition to a dict of its arrays with the node info
```

**Review by matthiasdiener — COMMENTED** [review]
Thanks!

**Review by kaushikcfd — APPROVED** [review]
Thanks. LGTM!

**Pull request merged**